tests/MongoDB.Driver.Tests/PipelineDefinitionBuilderTests.cs

```diff
@@ -29,6 +29,20 @@ namespace MongoDB.Driver.Tests
     public class PipelineDefinitionBuilderTests
     {
         // public methods
+        [Fact]
+        public void Create_should_throw_exception_when_output_type_does_not_match_last_stage_type()
+        {
+            var finalStageDefinition = new JsonPipelineStageDefinition<BsonDocument, BsonDocument>("{ project: { } } ");
+
+            var exception = Record.Exception(() => PipelineDefinition<BsonDocument, Starship>.Create(
+                stages: new IPipelineStageDefinition[] { finalStageDefinition }));
```

Comment:
With the right intermediate variables the Act stage can almost always be a one liner.

Reply:
New test uses a one liner :)